tests/bluetooth/tester/src/audio/btp_bap_broadcast.c

```diff
@@ -361,14 +427,115 @@ uint8_t btp_bap_broadcast_source_setup(const void *cmd, uint16_t cmd_len,
 	err = setup_broadcast_source(cp->streams_per_subgroup, cp->subgroups, source, &codec_cfg);
 	if (err != 0) {
 		LOG_DBG("Unable to setup broadcast source: %d", err);
+		return BTP_STATUS_FAILED;
+	}
+
+	/* Setup extended advertising data */
+	net_buf_simple_add_le16(&ad_buf, BT_UUID_BROADCAST_AUDIO_VAL);
+	net_buf_simple_add_le24(&ad_buf, broadcast_id);
+	base_ad[0].type = BT_DATA_SVC_DATA16;
+	base_ad[0].data_len = ad_buf.len;
+	base_ad[0].data = ad_buf.data;
+	base_ad[1].type = BT_DATA_NAME_COMPLETE;
+	base_ad[1].data_len = sizeof(CONFIG_BT_DEVICE_NAME) - 1;
+	base_ad[1].data = CONFIG_BT_DEVICE_NAME;
 
+	err = tester_gap_create_adv_instance(&ext_adv_param, BTP_GAP_ADDR_TYPE_IDENTITY,
+					     base_ad, ARRAY_SIZE(base_ad), NULL, 0, &gap_settings,
+					     &source->ext_adv);
+	if (err != 0) {
+		LOG_DBG("Failed to create extended advertising instance: %d", err);
 		return BTP_STATUS_FAILED;
 	}
 
-	err = bt_rand(&broadcast_id, BT_AUDIO_BROADCAST_ID_SIZE);
-	if (err) {
-		LOG_DBG("Unable to generate broadcast ID: %d\n", err);
+	err = tester_gap_padv_configure(source->ext_adv, &per_adv_param);
+	if (err != 0) {
+		LOG_DBG("Failed to configure periodic advertising: %d", err);
+		return BTP_STATUS_FAILED;
+	}
+
+	err = bt_bap_broadcast_source_get_base(source->bap_broadcast, &base_buf);
+	if (err != 0) {
+		LOG_DBG("Failed to get encoded BASE: %d\n", err);
+		return BTP_STATUS_FAILED;
+	}
 
+	per_ad = &per_adv_local[source->source_id][0];
+	per_ad->type = BT_DATA_SVC_DATA16;
+	per_ad->data_len = base_buf.len;
+	per_ad->data = base_buf.data;
+	err = tester_gap_padv_set_data(source->ext_adv, per_ad, 1);
+	if (err != 0) {
+		return BTP_STATUS_FAILED;
+	}
+
+	rp->gap_settings = gap_settings;
+	sys_put_le24(broadcast_id, rp->broadcast_id);
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+uint8_t btp_bap_broadcast_source_setup_v2(const void *cmd, uint16_t cmd_len,
+					  void *rsp, uint16_t *rsp_len)
+{
+	struct bt_le_per_adv_param per_adv_param =
+		*BT_LE_PER_ADV_PARAM(BT_GAP_MS_TO_PER_ADV_INTERVAL(150),
+				     BT_GAP_MS_TO_PER_ADV_INTERVAL(150), BT_LE_PER_ADV_OPT_NONE);
+	/* Zephyr Controller works best while Extended Advertising interval is a multiple
+	 * of the ISO Interval minus 10 ms (max. advertising random delay). This is
+	 * required to place the AUX_ADV_IND PDUs in a non-overlapping interval with the
+	 * Broadcast ISO radio events.
+	 */
+	struct bt_le_adv_param ext_adv_param =
+		*BT_LE_ADV_PARAM(BT_LE_ADV_OPT_EXT_ADV, BT_GAP_MS_TO_ADV_INTERVAL(140),
+				 BT_GAP_MS_TO_ADV_INTERVAL(140), NULL);
+	int err;
+	struct bt_audio_codec_cfg codec_cfg;
+	const struct btp_bap_broadcast_source_setup_v2_cmd *cp = cmd;
```

Comment:
when command is registered as 'variable length' no length checks are done in BTP generic code,  so some basic sanity checks should be done before accessing cp

(you may check other services for inspiration)